src/api/storage.py

```diff
@@ -327,25 +420,34 @@ def sliding_window_test(self, regressor, sliding_window, prior):
                     else:
                         # Use previous sample to shift window
                         self._data.loc[index][col_labels[2:]
-                                              ] = list(prev_sample.iloc[-2*sliding_window:-2])
+                                              ] = list(
+                                                  prev_sample.iloc[
+                                                      -2*sliding_window:-2
+                                                  ]
+                        )
 
-                        # Use regressor to predict and update first 2 positions in window
+                        # Use regressor to predict and update first 2 positions
                         self._data.loc[index][col_labels[0:2]
-                                              ] = regressor.predict([prev_sample])[0]
+                                              ] = regressor.predict(
+                                                  [prev_sample]
+                        )[0]
 
                     # Update prev_sample and prev_sample_index
                     prev_sample = self._data.loc[index]
                     prev_sample_index = index
         else:
             # Throw away data
             self._throw_data = True
+
             # Use different prior for each song
             song_num = 0
+
             for song_id, song in tqdm(self._data.groupby(level=0)):
                 prev_sample = []
                 prev_sample_index = 0
```

Comment:
The variable `prev_sample_index` is never used and can thus be removed. The variable `song_id` defined in the for-loop is also never used. In cases where the value must be handled such as in the for-loop, use the `_` notation i.e. `for _, song in tqdm(....):`. There are a few more places where unused variables are being declared - all of these could easily be fixed since they are indicated by the linter.